feat: add hotline reporting feature with modal and API endpoint

Justin322322 opened the pull request:
## Add Hotline Reporting Feature

Adds a reporting system for users to submit outdated hotline information directly from the Hotlines page. Reports are automatically created as GitHub issues.

### What's New
- Report modal with form validation and accessibility features
- `/api/report-hotline` endpoint that creates GitHub issues via API
- "Report Outdated Information" button on Hotlines page
- Privacy-first: optional email stored in server logs only, not in public issues

### Screenshots
<img width="1912" height="530" alt="image" src="https://github.com/user-attachments/assets/7dcb9fcf-f41c-4170-b72f-9df138e5558c" />

<img width="654" height="822" alt="image" src="https://github.com/user-attachments/assets/66c2647e-aaf8-490e-8d51-988f22a21f83" />



### Setup Required
Add `GITHUB_TOKEN` environment variable with `repo` scope to enable issue creation.

Justin322322 committed:
feat: add hotline reporting feature with modal and API endpoint

Justin322322 committed:
fix: improve security and reliability in report hotline

KishonShrill commented:
@Justin322322 Can I have an update on this? Does it still work on your end? I already used a GITHUB_TOKEN for it and gave me a "Network error. Please try again or report directly on GitHub."

KishonShrill commented:
Also, can you provide the steps on how I can test this? Kind of confused where to go after inputting the github token

Justin322322 committed:
refactor: replace GitHub integration with email-based reporting

- Remove GitHub API integration and token requirement
- Implement email-based reporting using MailChannels
- Send reports to bugs@bettergov.ph
- Make reporter email optional for anonymous reporting
- Simplify UI by removing 'Open in GitHub' button
- Add fallback console logging if email fails
- Update error messages to be more user-friendly

Justin322322 commented:
I've updated the hotline reporting feature to make it more accessible.

**Changes:**
- Removed the GitHub integration requirement
- Reports now go directly to bugs@bettergov.ph via email
- Email address is optional for anonymous reporting
- Simplified UI (removed "Open in GitHub" button)

**Technical Details:**
- Uses MailChannels API (free on Cloudflare Workers)
- Emails sent from: noreply@bettergov.ph
- Emails sent to: bugs@bettergov.ph
- API endpoint tested locally and working correctly (only logs)

**Testing Notes:**
- Email delivery requires deployment to Cloudflare Workers (MailChannels doesn't work on localhost)
- MailChannels may require domain verification or DNS records (SPF/DKIM) for bettergov.ph
- If MailChannels doesn't work after deployment, the code can easily be updated to use alternative services like Resend or SendGrid
- All report data is logged to console as fallback if email fails

**How to test after deployment (staging server):**
1. Go to /philippines/hotlines
2. Click "Report Outdated Information"
3. Fill in hotline name and issue (required fields)
4. Submit the form
5. Check bugs@bettergov.ph for the email

Justin322322 committed:
refactor: use GitHub issues instead of email for hotline reports

- Create public GitHub issues for transparency and volunteer visibility
- Users don't need GitHub accounts - API creates issues on their behalf
- Issues labeled with: hotline, data-update, community-report
- Reporter email hidden in HTML comment (only maintainers can see)
- Users receive link to view their submitted issue
- Improved error handling and user feedback

KishonShrill reviewed (approved):
Function tested and works 👍
